### DISCUSSION TITLE
Looking for  guidance on adding user registration functionality to a Vanna Flask application.

### DISCUSSION THREAD

**Discussion by anusreesoumya-nv** [vanna-ai/vanna#608] [Q&A] [answered]
Hi Team,
How do I add a new user to the Vanna Flask app, and what’s the recommended approach for handling user registration and login?

**Comment by E218PQ** [comment]
I have the same need

**Comment by zawalich** [comment]
@anusreesoumya-nv @E218PQ you have to write your own integration, see example in: https://vanna.ai/docs/web-app-auth/

**Reply by E218PQ** [reply]
I have tried, but unfortunately, has been unable to meet expectations.For example, the implementation of login verification, but the API is not available, can not do web page login verification and API token verification, I do not know if you have a similar case to share?

**Comment by zawalich** [comment, marked as answer]
@fireworks888 As I remember correctly, the provided example which I sent is
a mockup - and you have to write all integrations yourself, including those
you write about. It takes some doing, as well as OOP knowledge, but it is
the way. For my usecase, we needed SSO, and it works like a charm, as it is
custom built.
-----
Wysłane z urządzenia mobilnego.
Sent from a mobile device.
Pozdrowienia / Greetings
Aleksander Zawalich
https://zawalich.pl

śr., 26 lut 2025, 12:25 użytkownik fireworks888 ***@***.***>
napisał:
